src/azure-cli/azure/cli/command_modules/acr/check_health.py

```diff
@@ -418,28 +448,157 @@ def _check_private_endpoint(cmd, registry_name, vnet_of_private_endpoint):  # py
+
+# Validate if blob layer can be pulled
+def _check_blob(repository_name,
+                login_server,
+                username,
+                password,
+                image):
+
+    import requests
+    from ._docker_utils import get_manifest_authorization_header, parse_error_message
+
+    # Get manifest
+    # GET {url}/v2/{name}/manifests/{reference}
+    manifest_url = 'https://{}/v2/{}/manifests/{}'.format(login_server, repository_name, image)
+
+    manifest_response = requests.get(
+        manifest_url,
+        headers=get_manifest_authorization_header(username, password))
+
+    manifest = manifest_response.json()
+
+    # Get the digest of the smallest blob for performance purposes
+    smallest_blob = min(manifest['layers'], key=lambda layer: layer['size'])
+    digest = smallest_blob['digest']
+
+    # Get blob
+    request_url = _get_blob(login_server, repository_name, digest)
+    logger.debug(add_timestamp("Sending a HTTP GET request to {}".format(request_url)))
+
+    res = requests.get(request_url, auth=(username, password))
+
+    # Return server error message; if server error not available, will return error message below
+    if res.status_code >= 400:
+        raise CLIError(parse_error_message('Could not get the requested data.', res), res.status_code)
+    # Check if image or tag was inputted and return appropriate message
+    if res.status_code < 400 and 'sha256' not in image:
+        print_pass("Blobs referenced by tag '{}' can be pulled from registry '{}'".format(image, login_server))
+    else:
+        print_pass("Blobs in image '{}' can be pulled from registry '{}'".format(image, login_server))
 
 # General command
 def acr_check_health(cmd,  # pylint: disable useless-return
                      vnet=None,
                      ignore_errors=False,
                      yes=False,
-                     registry_name=None):
+                     registry_name=None,
+                     repository_name=None,
+                     image=None,
+                     resource_group_name=None):
     from azure.cli.core.util import in_cloud_console
+    from .repository import get_access_credentials
+    from ._utils import get_resource_group_name_by_registry_name
+
+    rg = get_resource_group_name_by_registry_name(cmd.cli_ctx, registry_name, resource_group_name)
+
+    registry = None
+
+    # Connectivity
+    try:
+        registry, _ = get_registry_by_name(cmd.cli_ctx, registry_name, rg)
+        login_server = registry.login_server.rstrip('/')
+    except CLIError:
+        from ._docker_utils import get_login_server_suffix
+        suffix = get_login_server_suffix(cmd.cli_ctx)
+
+        if not suffix:
+            from ._errors import LOGIN_SERVER_ERROR
+            _handle_error(LOGIN_SERVER_ERROR.format_error_message(registry_name), ignore_errors)
+            return
+
+    # Get the access credentials for the registry
+    login_server, username, password = get_access_credentials(
```

Comment:
isn't the pattern to add username and password for the costumer as option for this type?

Reply:
@rosanch sorry not clear on what your question means? So I am getting access credentials here and passing it down to the methods '_check_blob' and '_validate_image' so that I can use it in 'get_manifest_authorization_header' when I make the GET requests.

Reply:
I mean why do we not allow the costumer to pass their own credentials as an option?

Reply:
Yes good point, I will add username and password as an option, thank you!